app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb

```diff
@@ -71,7 +71,11 @@ def clusters
   def storagedomains
     collector.storagedomains.each do |storagedomain|
       storage_type = storagedomain.dig(:storage, :type).upcase
-      location = if storage_type == 'NFS' || storage_type == 'GLUSTERFS'
+      ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(storagedomain.try(:href))
+      location = case storage_type
+                 when 'LOCALFS'
+                   ems_ref
+                 when 'NFS', 'GLUSTERFS'
                    "#{storagedomain.dig(:storage, :address)}:#{storagedomain.dig(:storage, :path)}"
                  else
                    storagedomain.dig(:storage, :volume_group, :id)
```

Comment:
@borod108 do you know if this `else` case is really for all the other types of storage, or was it iSCSI and/or local block storage?

I'm wondering if we make `ems_ref` the "else" because it is guaranteed to at least work

Reply:
@agrare the "else" really does seem to work work for FCP as well (tested manually)
but in case we do not care about the location as much as before (which seems to be true with the latest changes) I would just change all of it to the ems_ref....

Reply:
Yeah we should try to keep the location the same for existing storages though.  I'm good with this then.

Reply:
If we can follow up with a more complete list of storage types and the ones where that volume_group.id doesn't work we can use href.

Reply:
According to the ovirt-engine api model we have the storage types described here:
https://ovirt.github.io/ovirt-engine-api-model/4.3/#types/storage_type

If I can say my opinion, to both match the intent of keeping the same location for existing storages and be "safe" in case of not well explored storages we can do something like the following:

 ```
location = case storage_type
                 when 'NFS', 'GLUSTERFS'
                   "#{storagedomain.dig(:storage, :address)}:#{storagedomain.dig(:storage, :path)}"
                 when 'ISCSI', 'FCP'
                   storagedomain.dig(:storage, :volume_group, :id)
                 else
                   ems_ref
```
Because I tried to query a glance storage type and it seems to me that it does not have more specific attributes than the LOCALFS one.

Reply:
:+1: that looks great, lets backport this simple fix and we can do this in a followup

Reply:
Ok perfect thx